zenguard/zenguard.py

```diff
@@ -73,18 +73,22 @@ def __init__(self, config: ZenGuardConfig):
             raise ValueError(f"LLM {config.llm} is not supported")
 
     def detect(self, detectors: list[Detector], prompt: str):
-        if len(detectors) == 0:
-            return {"error": "No detectors were provided"}
         try:
+            if detectors:
+                assert self.update_detectors(detectors) is None
```

Comment:
this has side effects that user might not want - the updated detectors.

we should be able to send the detectors along with the httpx request